deploy/examples/nfs.yaml

```diff
@@ -55,17 +56,3 @@ spec:
     #priorityClassName:
     # The logging levels: NIV_NULL | NIV_FATAL | NIV_MAJ | NIV_CRIT | NIV_WARN | NIV_EVENT | NIV_INFO | NIV_DEBUG | NIV_MID_DEBUG |NIV_FULL_DEBUG |NB_LOG_LEVEL
     logLevel: NIV_INFO
----
-apiVersion: ceph.rook.io/v1
```

Comment:
The .nfs pool isn't needed anymore?

Reply:
Rook should create the pool automatically. AFAIU, this is only necessary if users want to change the replication or some other such thing. I'll double check this to be sure though.

Reply:
I did revisit this. It seems like Rook doesn't create the pool automatically. How do we want to handle this? Should Rook require users to create the pool? If so, we will need to document it. I think it's probably easier if Rook creates it with the default configs if it doesn't already exist. This is conceptually similar to how the `device_health_metrics` and `.mgr` pools work from the viewpoint of users.

Reply:
So Ceph will create the pool automatically? If so, we could certainly follow the pattern of other default pools, and they can override the pool settings by creating the .nfs CephBlockPool. But then I'd also suggest keeping this example so they can see where to override the settings.

Reply:
Ceph doesn't create the pool automatically, no. I'm proposing that I think Rook should create the pool with default values if it doesn't exist. From the user standpoint the behavior is the same as if Ceph created it (like Ceph does for `.mgr`)

We still have the example in the `nfs-test.yaml`, but I don't see any reason to keep it in `nfs.yaml`. If users want to have the default pool size be replicas=2, then this example might trip them up.

Reply:
For Rook to create a pool automatically, we really need a spec that indicates how to create it. How about if we require them to create the CephBlockPool CR? Another approach would be to follow the example of CephObjectStore and CephFilesystem that embed the pool spec in their CRs, but not sure that approach makes sense for NFS?

Reply:
I don't think we do need a pool spec if we just do `ceph osd pool create` which just creates with Ceph's configured defaults. Which is effectively the same as ceph does with the `.mgr` pool. If users *want* to modify the pool, they can do so just the same as with the `.mgr` pool, but I don't think forcing them to create it is necessary or the best experience.

Reply:
Ok we could create the pool like that, but what does it look like if they create the .nfs CephBlockPool after that? We have to detect that the CR exists and skip that code to create it? It still feels simpler to me if we just keep it declarative and require the .nfs CephBlockPool to be specified, then there is no additional pool creation code needed at all.

Reply:
> Ok we could create the pool like that, but what does it look like if they create the .nfs CephBlockPool after that? We have to detect that the CR exists and skip that code to create it? It still feels simpler to me if we just keep it declarative and require the .nfs CephBlockPool to be specified, then there is no additional pool creation code needed at all.

@travisn @BlaineEXE 
I think maybe the following implementation can be used, PTAL.
This achieves all the goals discussed on this thread.

https://github.com/rook/rook/pull/10045

Reply:
If we required the .nfs CephBlockPool to be created, what is the isssue with the experience? They will be creating this nfs.yaml already, right? If it's included here, it will always be created so I don't see the issue with the experience.

Reply:
I'll be documenting the need for the `.nfs` pool in my changes today.

Reply:
We agreed to implement creation of the pool automatically. Behavior is noted here: https://github.com/rook/rook/pull/10054#discussion_r848723987